java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -267,6 +267,25 @@ public void dropColumns(List<String> columns) {
 
   private native void nativeDropColumns(List<String> columns);
 
+  /**
+   * Alter columns in the dataset.
+   *
+   * <p>For each column, there only support 4 keys in a map: - path: Path to the existing column to
+   * be altered. - rename: The new name of the column. If None, the column name will not be changed
+   * - nullable: Whether the column is nullable. If None, the nullability will not be changed. -
+   * data_type: The new data type of the column. If None, the data type will not be changed.
+   *
+   * @param columnAlterations the list of columns need to be altered.
+   */
+  public void alterColumns(List<Map<String, String>> columnAlterations) {
```

Comment:
Is this easy to use in Java? Only reason we use a dictionary in Python as parameters is Python users are used to passing dictionaries. But in other languages like Rust, we have a dedicated object for this ([rust docs](https://docs.rs/lance/latest/lance/dataset/struct.Dataset.html#method.alter_columns)). Are you sure a map makes sense here?

Reply:
> Is this easy to use in Java?

No.

I considered the design for this. I know there is a `ColumnAlteration`, so I extracted the map and passed it to it.

I did not define an object like `ColumnAlteration` in Java because I did not think the community would like to define many objects for these methods.

Ok, +1 for replacing map to object.

Reply:
@wjones127 Have addressed the review opinion, please have a look.